avaframe/ana1Tests/simiSolTest.py

```diff
@@ -662,7 +653,37 @@ def getSimiSolParameters(solSimi, relDict, ind_time, cfg):
 #########################
 
 
-def analyzeResults(particlesList, fieldsList, solSimi, relDict, cfg, outDirTest, index):
+def postProcessSimiSol(avalancheDir, cfgMain, cfgSimi, simDF, solSimi, outDirTest):
+    # loop on all the simulations and make the comparison to reference
+    for simHash, simDFrow in simDF.iterrows():
+        simName = simDFrow['simName']
+        # fetch the simulation results
+        particlesList, Tsave = com1DFA.readPartFromPickle(avalancheDir, simName=simName, flagAvaDir=True, comModule='com1DFA')
+        fieldsList, fieldHeader = com1DFA.readFields(avalancheDir, ['FD', 'FV', 'Vx', 'Vy', 'Vz'], simName=simName, flagAvaDir=True, comModule='com1DFA')
+        simDF.loc[simHash, 'Npart'] = particlesList[0]['Npart']
+        # analyze and compare results
+        hEL2Array, hELMaxArray, vEL2Array, vELMaxArray = analyzeResults(particlesList, fieldsList, solSimi, fieldHeader,
+                                                                        cfgSimi, outDirTest, simHash, simDFrow)
+        # add result of error analysis
+        # save results in the simDF
+        simDF.loc[simHash, 'hErrorL2'] = hEL2Array[-1]
+        simDF.loc[simHash, 'vErrorL2'] = vEL2Array[-1]
+        simDF.loc[simHash, 'hErrorLMax'] = hELMaxArray[-1]
+        simDF.loc[simHash, 'vErrorLMax'] = vELMaxArray[-1]
+        # +++++++++POSTPROCESS++++++++++++++++++++++++
+        # -------------------------------
+        # if cfgMain['FLAGS'].getboolean('showPlot'):
+        #     outAna1Plots.plotContoursSimiSol(particlesList, fieldsList, solSimi, relDict, cfgSimi, outDirTest)
+
+        outAna1Plots.showSaveTimeSteps(cfgMain, cfgSimi, particlesList, fieldsList, solSimi, Tsave, fieldHeader,
```

Comment:
see below too, I suggest to split this in 2 functions -> show and save. And put the show behind the showPlot flag (see above)

Reply:
I am not sure how I could do this. the plot saving is handled by the saveAndOrPlot function